File: MasterDirectory/MasterDirectory.Web/Modules/ServerTypes/TecnologiasInformacion/CategoriaTIForm.ts
import { StringEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface CategoriaTIForm {
    LocalSap: StringEditor;
    UsuarioGeo: StringEditor;
    Emaillocal: StringEditor;
    Extension: StringEditor;
    Telefono: StringEditor;
}

export class CategoriaTIForm extends PrefixedContext {
    static formKey = 'TecnologiasInformacion.CategoriaTI';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!CategoriaTIForm.init)  {
            CategoriaTIForm.init = true;

            var w0 = StringEditor;

            initFormType(CategoriaTIForm, [
                'LocalSap', w0,
                'UsuarioGeo', w0,
                'Emaillocal', w0,
                'Extension', w0,
                'Telefono', w0
            ]);
        }
    }
}

File: MasterDirectory/MasterDirectory.Web/Modules/TecnologiasInformacion/CategoriaTI/CategoriaTIForm.cs
using Serenity.ComponentModel;
using System;

namespace MasterDirectory.TecnologiasInformacion.Forms;

[FormScript("TecnologiasInformacion.CategoriaTI")]
[BasedOnRow(typeof(CategoriaTIRow), CheckNames = true)]
public class CategoriaTIForm
{
    [Tab("General")]
    public string LocalSap { get; set; }
    public string Nse { get; set; }
    public string Cluster { get; set; }
    public string Brick { get; set; }

    [Tab("GEO")]
    public string UsuarioGeo { get; set; }
    public string Emaillocal { get; set; }
    public string Extension { get; set; }
    public string Telefono { get; set; }
    
    //public DateTime DtRegistro { get; set; }
}